### PR TITLE
Rpc metrics in admin api

### PR DESCRIPTION
**Pull request by zajko**
Work done:
   * restructured metrics (moved to separate module)
   * added rest-api specific metrics (current connections, response times)
   * added db specific metrics (amount of data fetched from "raw" fields for events)
   * added rpc specific metrics
Above metrics are available in admin-api under the `/metrics` endpoint

### CODE REVIEW

**Review by jacek-casper — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -345,7 +345,7 @@ To monitor the Sidecar's memory consumption, observe the metrics page, `http://S
 process_resident_memory_bytes 292110336
 ```
 
-If memory consumption is high without an apparent reason, please inform the Sidecar team by creating an [issue in GitHub](https://github.com/CasperLabs/event-sidecar/issues).
+If memory consumption is high without an apparent reason, please inform the Sidecar team by creating an [issue in GitHub](https://github.com/casper-network/casper-sidecar//issues).
```
Comment (rafal-ch) [inline README.md]:
```suggestion
If memory consumption is high without an apparent reason, please inform the Sidecar team by creating an [issue in GitHub](https://github.com/casper-network/casper-sidecar/issues).
```

File: metrics/src/db.rs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                "db_raw_data_size",
+                "Size (in bytes) of raw data fetched from the database.",
+            ),
+            buckets: Vec::from(RAW_DATA_SIZE_BUCKETS as &'static [f64]),
+        },
+        &["message_type"],
+    )
+    .expect("db_raw_data_size metric can't be created");
+    REGISTRY
+        .register(Box::new(counter.clone()))
```
Comment (rafal-ch) [inline metrics/src/db.rs:22]: Is there a graceful shutdown use-case for the sidecar? If so, we might consider unregistering the metrics before the process is terminated. Not a big issue, though.

File: metrics/src/metrics.rs [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        return Err(MetricCollectionError::new(format!(
+            "error when encoding default prometheus metrics: {}",
+            e
+        )));
+    };
+    let res_custom = match String::from_utf8(buffer.clone()) {
+        Ok(v) => v,
+        Err(e) => {
+            return Err(MetricCollectionError::new(format!(
+                "Default and custom metrics have a non-utf8 character: {}",
```
Comment (rafal-ch) [inline metrics/src/metrics.rs]: a nit, but for consistency:
```suggestion
                "default and custom metrics have a non-utf8 character: {}",
```

File: sidecar/Cargo.toml [hunk 4]
```diff
@@ -6,7 +6,7 @@ edition = "2021"
 description = "Base module that spins up casper sidecar"
 readme = "README.md"
 homepage = "https://casperlabs.io"
-repository = "https://github.com/CasperLabs/event-sidecar/tree/dev"
+repository = "https://github.com/casper-network/casper-sidecar//tree/dev"
```
Comment (rafal-ch) [inline sidecar/Cargo.toml]:
```suggestion
repository = "https://github.com/casper-network/casper-sidecar/tree/dev"
```

**Review by rafal-ch — APPROVED** [review]
LGTM :+1:

**Review by zacshowa — APPROVED** [review]
LGTM

File: metrics/src/db.rs (hunk 2 above)
Comment (zajko) [reply metrics/src/db.rs:22]: We talked about this on slack. I don't recall any recommendations to deregister metrics before shutdown, but I'll revisit the issue.

Comment (rafal-ch) [reply metrics/src/db.rs:22]: All good, thanks :+1:

**Pull request merged**